src/main/java/com/exasol/adapter/document/UdfCallBuilder.java

```diff
@@ -83,31 +95,31 @@ private Select wrapStatementInStatementWithPostSelectionAndProjection(final List
      * statement.
      */
     private Select buildUdfCallStatement(final List<DocumentFetcher<DocumentVisitorType>> documentFetchers,
-            final RemoteTableQuery query, final String connectionName) throws IOException {
+            final RemoteTableQuery query) throws IOException {
         final Select doubleNestedSelect = StatementFactory.getInstance().select();
         final List<Column> emitsColumns = query
                 .getRequiredColumns().stream().map(column -> new Column(doubleNestedSelect,
                         column.getExasolColumnName(), convertDataType(column.getExasolDataType())))
                 .collect(Collectors.toList());
-        doubleNestedSelect.udf("Adapter." + ImportDocumentData.UDF_NAME, new ColumnsDefinition(emitsColumns),
-                column(DOCUMENT_FETCHER_PARAMETER), column(REMOTE_TABLE_QUERY_PARAMETER),
-                column(CONNECTION_NAME_PARAMETER));
-        final ValueTable valueTable = buildValueTable(documentFetchers, query, connectionName, doubleNestedSelect);
+        doubleNestedSelect.udf("Adapter." + ImportDocumentData.UDF_PREFIX + this.adapterName,
+                new ColumnsDefinition(emitsColumns), column(DOCUMENT_FETCHER_PARAMETER),
+                column(REMOTE_TABLE_QUERY_PARAMETER), column(CONNECTION_NAME_PARAMETER));
+        final ValueTable valueTable = buildValueTable(documentFetchers, query, doubleNestedSelect);
         doubleNestedSelect.from().valueTableAs(valueTable, "T", DOCUMENT_FETCHER_PARAMETER,
                 REMOTE_TABLE_QUERY_PARAMETER, CONNECTION_NAME_PARAMETER, FRAGMENT_ID);
         doubleNestedSelect.groupBy(column(FRAGMENT_ID));
         return doubleNestedSelect;
     }
 
     private ValueTable buildValueTable(final List<DocumentFetcher<DocumentVisitorType>> documentFetchers,
-            final RemoteTableQuery query, final String connectionName, final Select select) throws IOException {
+            final RemoteTableQuery query, final Select select) throws IOException {
         final ValueTable valueTable = new ValueTable(select);
         int rowCounter = 0;
         for (final DocumentFetcher<DocumentVisitorType> documentFetcher : documentFetchers) {
             final String serializedDocumentFetcher = StringSerializer.serializeToString(documentFetcher);
             final String serializedRemoteTableQuery = StringSerializer.serializeToString(query);
             final ValueTableRow row = ValueTableRow.builder(select).add(serializedDocumentFetcher)
-                    .add(serializedRemoteTableQuery).add(connectionName).add(rowCounter++).build();
+                    .add(serializedRemoteTableQuery).add(this.connectionName).add(rowCounter++).build();
             valueTable.appendRow(row);
         }
```

Comment:
```suggestion
                    .add(serializedRemoteTableQuery) //
                    .add(this.connectionName) //
                    .add(rowCounter) //
                    .build();
            valueTable.appendRow(row);
            ++rowCounter;
        }
```

Use side effects only if really necessary.